src/main/java/neqsim/thermodynamicoperations/flashops/Flash.java

```diff
@@ -397,36 +397,35 @@ public void solidPhaseFlash() {
         tempVar[k] = system.getPhase(0).getComponent(k).getz();
         for (int i = 0; i < system.getNumberOfPhases() - 1; i++) {
           tempVar[k] -=
-              system.getBeta(i) * system.getPhases()[3].getComponent(k).getFugacityCoefficient()
+              system.getBeta(i) * system.getPhase(3).getComponent(k).getFugacityCoefficient()
                   / system.getPhase(i).getComponent(k).getFugacityCoefficient();
         }
 
         if (tempVar[k] > 0.0 && tempVar[k] > frac) {
           solidPhase = true;
           solid = k;
           frac = tempVar[k];
-          for (int p = 0; p < system.getPhases()[0].getNumberOfComponents(); p++) {
-            system.getPhases()[3].getComponent(p).setx(1.0e-20);
+          for (int p = 0; p < system.getPhase(0).getNumberOfComponents(); p++) {
+            system.getPhase(3).getComponent(p).setx(1.0e-20);
           }
-          system.getPhases()[3].getComponents()[solid].setx(1.0);
+          system.getPhase(3).getComponents()[solid].setx(1.0);
         }
         // logger.info("tempVar: " + tempVar[k]);
       }
     }
 
     if (solidPhase) {
-      if (frac < system.getPhases()[0].getComponents()[solid].getz() + 1e10) {
+      if (frac < system.getPhase(0).getComponents()[solid].getz() + 1e10) {
         for (int i = 0; i < system.getNumberOfPhases() - 1; i++) {
-          // system.getPhases()[i].getComponents()[solid].setx(1.0e-10);
+          // system.getPhase(i).getComponents()[solid].setx(1.0e-10);
         }
         system.init(1);
         // logger.info("solid phase will form..." + system.getNumberOfPhases());
         // logger.info("freezing component " + solid);
         system.setBeta(system.getNumberOfPhases() - 1, frac);
         system.initBeta();
         system.setBeta(system.getNumberOfPhases() - 1,
-            system.getPhases()[3].getComponent(solid).getNumberOfmoles()
-                / system.getNumberOfMoles());
+            system.getPhase(3).getComponent(solid).getNumberOfmoles() / system.getNumberOfMoles());
```

Comment:
In this case I beleive we have to used system.getPhases()[3] - as the solid phase is always number three (and getPhase(3) can point to a different phase.